mono/metadata/threadpool.c

```diff
@@ -350,9 +351,9 @@ worker_callback (void)
 
 		domains_unlock ();
 
-		MonoString *thread_name = mono_string_new_checked (mono_get_root_domain (), "Thread Pool Worker", &error);
+		MonoStringHandle thread_name = mono_string_new_handle (mono_get_root_domain (), "Thread Pool Worker", &error);
 		mono_error_assert_ok (&error);
-		mono_thread_set_name_internal (thread, thread_name, FALSE, TRUE, &error);
+		mono_thread_set_name_handle_internal (thread, thread_name, FALSE, TRUE, &error);
```

Comment:
Yea, I really think we just need a version of `set_name` that takes a `const char*` and uses the handle version internally.
(otherwise this again leaks a handle due to the missing prolog/epilog macros)

Reply:
Agreed and there is already the utf8 version. I can allow negative length also and call strlen.